deephyper/skopt/optimizer/optimizer.py

```diff
@@ -651,10 +677,16 @@ def _filter_failures(self, yi):
         if self.filter_failures in ["mean", "max"]:
             yi_no_failure = [v for v in yi if v != OBJECTIVE_VALUE_FAILURE]
 
-            if self.filter_failures == "mean":
-                yi_failed_value = np.mean(yi_no_failure)
+            # when yi_no_failure is empty all configurations are failures
+            if len(yi_no_failure) == 0:
+                if len(yi) == self.max_failures:
```

Comment:
condition `len(yi) >= self.max_failures` is better as multiple objectives can be added at once.

Reply:
Sure